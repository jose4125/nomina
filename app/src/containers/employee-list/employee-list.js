import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

class EmployeeList extends Component {
  renderList () {
    return this.props.employees.map(employee => {
      return (
        <li key={employee.name}>{employee.name}</li>
      )
    })
  }

  render () {
    return (
      <ul>
        {this.renderList()}
      </ul>
    )
  }
}

EmployeeList.propTypes = {
  employees: PropTypes.array
}

function mapStateToProps (state) {
  return {
    employees: state.employees
  }
}

export const Unwrapped = EmployeeList

export default connect(mapStateToProps)(EmployeeList)
