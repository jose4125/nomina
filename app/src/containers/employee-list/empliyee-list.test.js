import React from 'react'
import {Unwrapped as UnwrappedEmployeeList} from './employee-list'
import employees from '../../reducers/employees-reducer'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

test('Employee list >> App snapshot test', () => {
  const employeesArr = employees()
  const component = shallow(<UnwrappedEmployeeList employees={employeesArr} />)
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})
