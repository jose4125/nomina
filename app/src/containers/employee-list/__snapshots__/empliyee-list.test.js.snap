// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employee list >> App snapshot test 1`] = `
<ul>
  <li>
    jose david
  </li>
  <li>
    diana
  </li>
</ul>
`;
